### ISSUE TITLE
adding update support for labels on google_logging_metric 

### ISSUE THREAD

**Issue by enricojonas** [hashicorp/terraform-provider-google#9921]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version:
```
Terraform v1.0.5
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v3.81.0
```

### Affected Resource(s)

* google_logging_metric

### Terraform Configuration Files

Before

```tf
resource "google_logging_metric" "test_metric" {
  project = var.gcp_project
  name    = "testMetric"


  filter = <<-EOT
        resource.type="cloud_run_revision"
    EOT

  metric_descriptor {
    metric_kind = "DELTA"
    unit        = "1"
    value_type  = "INT64"

    labels {
      key        = "operationName"
      value_type = "STRING"
    }

  }

  label_extractors = {
    "operationName" = "EXTRACT(jsonPayload.apiOperationName)"
  }

}
```

After adding an additional metric_descriptor label and label_extractor "clientApplicationName"

```tf
resource "google_logging_metric" "test_metric" {
  project = var.gcp_project
  name    = "testMetric"


  filter = <<-EOT
        resource.type="cloud_run_revision"
    EOT

  metric_descriptor {
    metric_kind = "DELTA"
    unit        = "1"
    value_type  = "INT64"

    labels {
      key        = "operationName"
      value_type = "STRING"
    }

    labels {
      key        = "clientApplicationName"
      value_type = "STRING"
    }
  }

  label_extractors = {
    "operationName" = "EXTRACT(jsonPayload.apiOperationName)"
    "clientApplicationName"    = "EXTRACT(jsonPayload.clientApplicationName)"
  }

}
```

### Debug Output

N/A

### Panic Output

N/A

### Expected Behavior

Terraform should update the resource in-place. This will preserve historical data for this metric. You can modify the metric in the UI, preserving the data.

### Actual Behavior

Terraform is re-creating the resource and the metric's historical data is lost.

```tf
Terraform will perform the following actions:

  # google_logging_metric.test_metric must be replaced
-/+ resource "google_logging_metric" "test_metric" {
      ~ id               = "testMetric" -> (known after apply)
      ~ label_extractors = {
          + "clientApplicationName" = "EXTRACT(jsonPayload.clientApplicationName)"
            # (1 unchanged element hidden)
        }
        name             = "testMetric"
        # (2 unchanged attributes hidden)

      ~ metric_descriptor {
            # (3 unchanged attributes hidden)

          + labels { # forces replacement
              + key        = "clientApplicationName"
              + value_type = "STRING"
            }
            # (1 unchanged block hidden)
        }
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```

### Steps to Reproduce

1. Create a `google_logging_metric` resource
2. Attempt to add a label_extractor and metric_descriptor label as in above example config
3. Observe the provider error as above

### Important Factoids

N/A

### References

N/A

**Comment by melinath** [comment]
Note that the API [doesn't actually support an "update" operation](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.metricDescriptors/create), but it seems like "creating" something that already exists sort of updates it? This is how we currently handle updating the labels field to e.g. add new labels

**Comment by melinath** [comment]
b/239950421

**Comment by mattalbr** [comment]
FWIW, calling this an "enhancement" seems disingenuous to me. I think an accurate description of the bug would be
"google_cloud_logging loses historical data upon adding new labels".

**Comment by tm185187** [comment]
This is an issue, from console you can add a labels after creating a metric, but with terraform we are unable to add a label after creation of log based metric, this leads to metric update in place. This leads to a large issue if any metric is referenced by an alerting policy, it will not let you delete it.

**Comment by mouyang** [comment]
> Note that the API [doesn't actually support an "update" operation](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.metricDescriptors/create), but it seems like "creating" something that already exists sort of updates it? This is how we currently handle updating the labels field to e.g. add new labels

+1 @tm185187 

@enricojonas isn't trying to update the resource (MetricDescriptor) in the link above directly.  They are trying to add a label to an existing LogMetric's MetricDescriptor (https://cloud.google.com/logging/docs/reference/v2/rest/v2/projects.metrics), and the LogMetric does support the update operation.  I was able to add a label through the update operation in the API and therefore I should be able to do this through Terraform as well.

**Comment by tm185187** [comment]
@mouyang Nope, GCP API does support update operation, but Terraform doesn't support. You can reproduce in your local and let us know!

**Comment by roaks3** [comment]
This is a duplicate of https://github.com/hashicorp/terraform-provider-google/issues/14998. There is recent discussion on both issues, so we can keep this open for reference until it is resolved, but please try post any new comments on https://github.com/hashicorp/terraform-provider-google/issues/14998 (it is being actively looked at by the service team).
